### ISSUE TITLE
MDC is empty after a Jaeger sub span is closed.

### ISSUE THREAD

**Issue by KeesKoffeman** [quarkusio/quarkus#8722]
**Describe the bug**
After a subspan closes MDC is empty.

**Expected behavior**
MDC context should contain traceId, spanId and sample of parent span.

**Actual behavior**
The MDC context is wiped on MDCScope.close() without restoring the MDC info the parent span.

**To Reproduce**
1. Create Trace A 
2. Within Trace A create Trace B
3. Within Trace A inspect MDC after Trace B closes
4. MDC is empty.

**Comment by gsmet** [comment]
@KeesKoffeman any chance you could create a reproducer?

/cc @objectiser

**Comment by Ladicek** [comment]
I wonder if this could be caused by one thing I noticed in https://github.com/quarkusio/quarkus/blob/master/extensions/jaeger/runtime/src/main/java/io/quarkus/jaeger/runtime/MDCScopeManager.java: the `activate` method returns `MDCScope` (wrapping the underlying scope), but the `active` method doesn't (no wrapping occurs there).

CC @pavolloffay

**Comment by KeesKoffeman** [comment]
```
import io.jaegertracing.internal.JaegerSpanContext;
import io.opentracing.Scope;
import io.opentracing.Span;
import io.opentracing.SpanContext;
import io.opentracing.util.ThreadLocalScopeManager;
import io.quarkus.jaeger.runtime.MDCScopeManager;
import org.junit.jupiter.api.Test;
import org.slf4j.MDC;

import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class MDCScopeBugReproducer {

    @Test
    public void proveMdcIsEmpty() {

        JaegerSpanContext span = new JaegerSpanContext(1, 1, 1, 1, Byte.valueOf("0"));
        JaegerSpanContext subSpan = new JaegerSpanContext(2, 2, 2, 1, Byte.valueOf("0"));

        ThreadLocalScopeManager threadLocalScopeManager = new ThreadLocalScopeManager();
        MDCScopeManager mdcScopeManager = new MDCScopeManager(threadLocalScopeManager);

        Scope scope = mdcScopeManager.activate(new TestSpan(span), true);
        assertEquals(span, threadLocalScopeManager.active().span().context());
        assertEquals("10000000000000001", MDC.get("traceId"));

        Scope subScope = mdcScopeManager.activate(new TestSpan(subSpan), true);
        assertEquals(subSpan, threadLocalScopeManager.active().span().context());
        assertEquals("20000000000000002", MDC.get("traceId"));

        subScope.close();

        // After closing the subSpan the ThreadLocalScope restores the previous scope.
        assertEquals(span, threadLocalScopeManager.active().span().context());

        // But the MDCScope does not restore the previous MDC.
        assertEquals("10000000000000001", MDC.get("traceId"));
    }

    class TestSpan implements Span {

        private final SpanContext spanContext;

        TestSpan(SpanContext spanContext) {
            this.spanContext = spanContext;
        }

        @Override public SpanContext context() { return spanContext; }
        @Override public Span setTag(String key, String value) { return this; }
        @Override public Span setTag(String key, boolean value) { return this; }
        @Override public Span setTag(String key, Number value) { return this; }
        @Override public Span log(Map<String, ?> fields) { return this; }
        @Override public Span log(long timestampMicroseconds, Map<String, ?> fields) { return this; }
        @Override public Span log(String event) { return this; }
        @Override public Span log(long timestampMicroseconds, String event) { return this; }
        @Override public Span setBaggageItem(String key, String value) { return this; }
        @Override public String getBaggageItem(String key) { return null; }
        @Override public Span setOperationName(String operationName) { return this; }
        @Override public void finish() { }
        @Override public void finish(long finishMicros) { }
    }
}
```

**Comment by KeesKoffeman** [comment]
I stumbled upon this while implementing `quarkus-logging-gelf` in combination with `quarkus-smallrye-opentracing`. Due to this issue I ended up with log entries which are missing the correlation to their trace.

**Comment by gsmet** [comment]
@pavolloffay @Ladicek any chance we could have a fix for that today? It would be great to have it fixed in 1.4.

**Comment by Ladicek** [comment]
I have a fix in #8746, but I'd really appreciate some OpenTracing expert to review. Apparently the MDC scope thing was created by @pavolloffay, so if he could review, that would be great :-)
